app/controllers/api/v1/merchants_controller.rb

```diff
@@ -0,0 +1,28 @@
+class Api::V1::MerchantsController < ApplicationController
+  def index
+    render json: MerchantSerializer.new(Merchant.all)
+  end
+
+  def show
+    render json: MerchantSerializer.new(Merchant.find(params[:id]))
+  end
+
+  def create
+    render json: MerchantSerializer.new(Merchant.create(merchant_params))
+  end
+
+  def update
+    render json: MerchantSerializer.new(Merchant.update(params[:id], merchant_params))
+  end
+
+  def destroy
+    merchant = Merchant.find(params[:id])
+    merchant.destroy_merchant_and_associations
```

Comment:
With no JSON return, generates 204 No Content status code. This is recommended by the JSON API Specification